x/evm/abci.go

```diff
@@ -578,28 +578,23 @@ func handleGeneralMessage(ctx sdk.Context, ck types.ChainKeeper, n types.Nexus,
 	return cmd.ID, nil
 }
 
-func handleGeneralMessages(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus, m types.MultisigKeeper) {
+func handleMessages(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus, m types.MultisigKeeper) {
 
 	for _, chain := range slices.Filter(n.GetChains(ctx), types.IsEVMChain) {
 		ck := funcs.Must(bk.ForChain(ctx, chain.Name))
 		endBlockerLimit := ck.GetParams(ctx).EndBlockerLimit
 		msgs := n.GetApprovedMessages(ctx, chain.Name, endBlockerLimit)
-		bk.Logger(ctx).Debug(fmt.Sprintf("handling %x general messages", len(msgs)), types.AttributeKeyChain, chain.Name)
+
+		bk.Logger(ctx).Debug(fmt.Sprintf("handling %d general messages", len(msgs)), types.AttributeKeyChain, chain.Name)
 
 		for _, msg := range msgs {
-			success := utils.RunCached(ctx, bk, func(ctx sdk.Context) (bool, error) {
-				cmdID, err := handleGeneralMessage(ctx, ck, n, m, chain, msg)
+			success := false
+			capturedErr := fmt.Errorf("unknown")
+			_ = utils.RunCached(ctx, bk, func(ctx sdk.Context) (bool, error) {
+				cmdID, err := handleMessage(ctx, ck, n, m, chain, msg)
 				if err != nil {
-					ck.Logger(ctx).Debug(fmt.Sprintf("failed handling general message: %s", err.Error()),
-						types.AttributeKeyChain, chain.Name.String(),
-						types.AttributeKeyMessageID, msg.ID.ID,
-					)
-					events.Emit(ctx, &types.ContractCallFailed{
-						Chain:     chain.Name,
-						MessageID: msg.ID.ID,
-						Reason:    err.Error(),
-					})
-
+					capturedErr = err
```

Comment:
you don't need to capture the error. RunCached will log the error being returned here. We can just log that the message handling failed below